package/Makefile

```diff
@@ -132,7 +136,7 @@ ifneq ($(CONFIG_USE_APK),)
 			--sign $(BUILD_KEY_APK_SEC) \
 			--output packages.adb \
 			$$(ls *.apk | grep -vE '^(base-files-|kernel-|libc-)'); \
-		echo -n '{"architecture": "$(ARCH_PACKAGES)", "packages":{' > index.json; \
+		echo -n '{"architecture": "$(ARCH_PACKAGES)", "kernel": "$(KERNEL_VERSION)", "packages":{' > index.json; \
```

Comment:
I might be missing something, but why do you need to put this information into every index.json? IMO its not related to packages, but rather to a target, so shouldn't this be provided in profiles.json instead?

```diff
diff --git a/scripts/json_overview_image_info.py b/scripts/json_overview_image_info.py
index 0d2cf7f1ef51..6161704bc890 100755
--- a/scripts/json_overview_image_info.py
+++ b/scripts/json_overview_image_info.py
@@ -47,7 +47,7 @@ for device_id, profile in output.get("profiles", {}).items():
 
 
 if output:
-    default_packages, output["arch_packages"] = run(
+    default_packages, output["arch_packages"], output["linux_kernel"] = run(
         [
             "make",
             "--no-print-directory",
@@ -55,6 +55,7 @@ if output:
             "target/linux/",
             "val.DEFAULT_PACKAGES",
             "val.ARCH_PACKAGES",
+            "val.LINUX_KERNEL:=$(LINUX_VERSION)-$(LINUX_RELEASE)-$(LINUX_VERMAGIC)",
             "V=s",
         ],
         stdout=PIPE,
```

Reply:
The above would be again, only ASU specific variable. To have it tool agnostic, it should be perhaps

```diff
diff --git a/scripts/json_overview_image_info.py b/scripts/json_overview_image_info.py
index 0d2cf7f1ef51..96921c27430a 100755
--- a/scripts/json_overview_image_info.py
+++ b/scripts/json_overview_image_info.py
@@ -47,7 +47,13 @@ for device_id, profile in output.get("profiles", {}).items():
 
 
 if output:
-    default_packages, output["arch_packages"] = run(
+    (
+        default_packages,
+        output["arch_packages"],
+        linux_version,
+        linux_release,
+        linux_vermagic,
+    ) = run(
         [
             "make",
             "--no-print-directory",
@@ -55,6 +61,9 @@ if output:
             "target/linux/",
             "val.DEFAULT_PACKAGES",
             "val.ARCH_PACKAGES",
+            "val.LINUX_VERSION",
+            "val.LINUX_RELEASE",
+            "val.LINUX_VERMAGIC",
             "V=s",
         ],
         stdout=PIPE,
@@ -64,7 +73,11 @@ if output:
     ).stdout.splitlines()
 
     output["default_packages"] = sorted(default_packages.split())
-
+    output["linux_kernel"] = {
+        "version": linux_version,
+        "release": linux_release,
+        "vermagic": linux_vermagic,
+    }
     output_path.write_text(json.dumps(output, sort_keys=True, separators=(",", ":")))
 else:
     print("JSON info file script could not find any JSON files for target")

```

Reply:
> IMO its not related to packages, but rather to a target

Oh, absolutely agree and very much like your last example, but...

I actually briefly considered putting it in profiles, and I very much like it from a structural and elegance point of view.  But, I wanted to reduce the number of downloads/connections/bandwidth required, so stuffed it into the package index for practical considerations.  I don't know how valid an argument that is, so let me know what you think.

Old way (23.05 and earlier with kmods in target package index), single download:
```
GET .../packages/index.json 
```

This PR, two downloads:
```
GET .../packages/index.json
GET .../kmods/<kernel-from-index>/index.json
```

Your proposed kernel info in profiles, three downloads:
```
GET .../profiles.json
GET .../packages/index.json
GET .../kmods/<kernel-from-profiles>/index.json
```

The issue I saw was that `profiles.json` can be relatively huge (a mere 3k for x86, but 368k -ouch!- right now for ath79[1]), does that matter in the greater scheme of things?  If not, I much prefer your last example with the components broken out into their constituent parts.

(Or maybe add another file `target.json` splitting or subsetting current `profiles.json` "header" data?)

[1] https://downloads.openwrt.org/snapshots/targets/ath79/generic/profiles.json

Reply:
Ok, I should give a quick summary of the two use cases I'm considering:
1) Easy one first: OpenWrt-based clients (`owut` specifically, might add to LuCI app) want to know kver = '6.6.61' or whatever to be able to say "time to upgrade".
2) ASU server's existing api: `GET .../json/v1/snapshots/targets/ath79/generic/index.json`, so clients can compare the upstream version numbers with installed versions and present a "time to upgrade" verdict.

For 1, data can live anywhere, we're downloading a ton of files anyhow, so my bandwidth arguments are moot.

For 2, we only use the "package":"version" pairs, so anything else is noise and overhead (which includes virtually everything in `profiles.json`).

Reply:
> For 1, data can live anywhere, we're downloading a ton of files anyhow, so my bandwidth arguments are moot.

Oh, I should also add that for this use case having the kernel data in `profiles.json` would be much preferred.  Trying to extract it from the package indexes violates my "module boundaries" in `owut`, so that's another argument for your proposed solution.

Reply:
@efahl I would just put it in `profiles.json`, to me that makes the most sense

Reply:
@robimarko Yeah, me too.  But, just to be sure, you're not concerned with extra bandwidth usage?

As I recall, we get maybe 4-500 download requests on the ASU server per day.  Assuming those are all as big as ath79, then our worst case is an extra ~180 MB coming out of `downloads` and into `sysupgrade`...

Reply:
I doubt that will be the deal breaker compared to all of the regular downloads and AI scrapers

Reply:
Ah, yes, our friends the friggin' AI scrapers...

@ynezz , If I drop this PR in lieu of yours, do you want just commit your changes, or should I copy your code, submit a new PR and add you as 'Authored-by:' or something on the new one?

Reply:
Oh, and before I forget, it needs to get into 24.10, too.

Reply:
> If I drop this PR in lieu of yours, do you want just commit your changes, or should I copy your code, submit a new PR and add you as 'Authored-by:' or something on the new one?

Please don't drop anything, its a simple "oneliner" suggestion, so if you're OK with the change, simply take it, no need for the credits.

Reply:
Is 0b572fcc61c7f3 fine with you?

Reply:
Perfect, thanks!